lib/utils/getSelectorMeaningfulSource.mjs

```diff
@@ -0,0 +1,45 @@
+import selectorParser from 'postcss-selector-parser';
+
+/**
+ * Get the source index of a selector, excluding leading whitespace.
+ *
+ * @param {import('postcss-selector-parser').Selector} node
+ *
+ * @returns {{selectorStr: string, index: number, endIndex: number}}
+ */
+export default function getSelectorMeaningfulSource(node) {
+	const firstIndex = node.nodes.findIndex((x) => !selectorParser.isComment(x));
+	const lastIndex = node.nodes.findLastIndex((x) => !selectorParser.isComment(x));
+
+	const first = node.nodes[firstIndex];
+	let last = node.nodes[lastIndex];
+
+	if (!first || !last) {
+		const selectorStr = node.toString().trim();
+		const index = node.sourceIndex ?? 0;
+		const endIndex = index + selectorStr.length;
+
+		return {
+			selectorStr,
+			index,
+			endIndex,
+		};
+	}
+
+	last = last.rawSpaceAfter.length ? last.clone() : last;
+	last.rawSpaceAfter = '';
```

Comment:
I suppose there's no need to set `rawSpaceBefore`?

Reply:
It is important not to alter this.
We only want to trim then endings, we don't want to alter the middle.

Altering `rawSpaceBefore` for the last node means that we would alter the middle part of a selector.